pkg/types/validation/installconfig.go

```diff
@@ -996,5 +1008,91 @@ func validateFeatureSet(c *types.InstallConfig) field.ErrorList {
+				if !validateLoadBalancer(c.BareMetal.LoadBalancer.Type) {
+					allErrs = append(allErrs, field.Invalid(field.NewPath("platform", "baremetal", "loadBalancer", "type"), c.VSphere.LoadBalancer.Type, "invalid load balancer type"))
+				}
+				if c.BareMetal.LoadBalancer.Type == configv1.LoadBalancerTypeOpenShiftManagedDefault && len(c.BareMetal.APIVIPs) == 0 {
+					allErrs = append(allErrs, field.Required(field.NewPath("platform", "baremetal", "apiVIPs"), "must be set when using the OpenShiftManaged default load balancer"))
+				}
+			}
+		}
+		if c.Nutanix != nil {
+			if c.Nutanix.LoadBalancer != nil {
+				if !validateLoadBalancer(c.Nutanix.LoadBalancer.Type) {
+					allErrs = append(allErrs, field.Invalid(field.NewPath("platform", "nutanix", "loadBalancer", "type"), c.Nutanix.LoadBalancer.Type, "invalid load balancer type"))
+				}
+				if c.Nutanix.LoadBalancer.Type == configv1.LoadBalancerTypeOpenShiftManagedDefault && len(c.Nutanix.APIVIPs) == 0 {
+					allErrs = append(allErrs, field.Required(field.NewPath("platform", "nutanix", "apiVIPs"), "must be set when using the OpenShiftManaged default load balancer"))
+				}
+			}
+		}
+		if c.OpenStack != nil {
+			if c.OpenStack.LoadBalancer != nil {
+				if !validateLoadBalancer(c.OpenStack.LoadBalancer.Type) {
+					allErrs = append(allErrs, field.Invalid(field.NewPath("platform", "openstack", "loadBalancer", "type"), c.OpenStack.LoadBalancer.Type, "invalid load balancer type"))
+				}
+				if c.OpenStack.LoadBalancer.Type == configv1.LoadBalancerTypeOpenShiftManagedDefault && len(c.OpenStack.APIVIPs) == 0 {
+					allErrs = append(allErrs, field.Required(field.NewPath("platform", "openstack", "apiVIPs"), "must be set when using the OpenShiftManaged default load balancer"))
+				}
+			}
+		}
+		if c.Ovirt != nil {
+			if c.Ovirt.LoadBalancer != nil {
+				if !validateLoadBalancer(c.Ovirt.LoadBalancer.Type) {
+					allErrs = append(allErrs, field.Invalid(field.NewPath("platform", "ovirt", "loadBalancer", "type"), c.Ovirt.LoadBalancer.Type, "invalid load balancer type"))
+				}
+				if c.Ovirt.LoadBalancer.Type == configv1.LoadBalancerTypeOpenShiftManagedDefault && len(c.Ovirt.APIVIPs) == 0 {
+					allErrs = append(allErrs, field.Required(field.NewPath("platform", "ovirt", "apiVIPs"), "must be set when using the OpenShiftManaged default load balancer"))
+				}
+			}
+		}
+		if c.VSphere != nil {
+			if c.VSphere.LoadBalancer != nil {
+				if !validateLoadBalancer(c.VSphere.LoadBalancer.Type) {
+					allErrs = append(allErrs, field.Invalid(field.NewPath("platform", "vsphere", "loadBalancer", "type"), c.VSphere.LoadBalancer.Type, "invalid load balancer type"))
+				}
+				if c.VSphere.LoadBalancer.Type == configv1.LoadBalancerTypeOpenShiftManagedDefault && len(c.VSphere.APIVIPs) == 0 {
+					allErrs = append(allErrs, field.Required(field.NewPath("platform", "vsphere", "apiVIPs"), "must be set when using the OpenShiftManaged default load balancer"))
+				}
+			}
+		}
```

Comment:
These validations should probably be moved to the platform-specific validations ([example](https://github.com/openshift/installer/blob/master/pkg/types/openstack/validation/platform.go) or at least they shouldn't be coupled with the featureset validations. 

There should be one validation to enforce that the field can only be set when tech preview is enabled, which is what you have in the following `else` block. The rest of the validations don't need to worry about tech preview, because they should only be run when a value is set.

Reply:
I've put them here because vsphere had some validations for the Zones work (that have been removed in the meantime).
I'll move them per platform then.